src/common/options.c

```diff
@@ -49,3 +52,122 @@ void _set_seed_default(tinymt64_t *random_state, const bool choose_seed,
     tinymt64_init_by_array(random_state, init_key, key_length);
   }
 }
+
+/* Simple set_seed function for the basic generators */
```

Comment:
Simplify all the functions by replacing the calls with a single random generator.